library/src/main/java/com/whatsapp/stringpacks/StringPacks.java

```diff
@@ -203,7 +204,8 @@ private static boolean useSystemResources(@NonNull Locale locale) {
   }
 
   private static String getPackFileName(Locale locale) {
-    return "strings_" + locale.getLanguage();
+    return String.format("strings_%s" + (!locale.getCountry().equals("") ? "-r%s" : ""),
```

Comment:
For the same reason that people may want to pack multiple locales into one pack file, like `es-MX`, `es-ES`, `es` into `strings_es.pack` file via `pack_id_mapping` property, we can not hard-code this logic here.

We should again expose an API for apps to provide proper pack file name that can be appended to `strings_` based on their implementation.

```suggestion
    String packFileId = null;
    if (stringPackLocaleManager != null) {
        packFileId = stringPackLocaleManager.getPackFileIdForLocale(locale);
    }
    if (packFileId == null) {
        packFileId = locale.getLanguage();
    }
    return "strings_" + packFileId;
```

Reply:
Yes, totally makes sense.
So, the stringPackLocaleManager should have access to the `"pack_id_mapping" : {}` that is defined in the `config.json` right? 
And if a mapping is provided for `es-MX` to `es` in `config.json` then `stringPackLocaleManager.getPackFileIdForLocale(es-MX)` would return `es` otherwise `null`.

Any suggestions on how the library will have access on the config file? Or is there a better approach?

Reply:
Thoughts on this approach:

Similar to how the `assemble_string_packs.py` script generates `StringPackIds.java` we also generate `StringPacksLocaleManger.java` which would contain the `pack_id_mapping` from the `config.json`
During the application setup we would need to do something like `StringPacksLocaleManager.registerLocaleManger()`. This would give us access to the mapping locales defined by the user.

Reply:
@akshob I think it gives the developers more flexibility if this is not a generated file but a file created by application developers and then registered via the API.
Applications can call `StringPacks.registerStringPackLocaleManager()` inside of their Application class' `attachBaseContext()` method.
In the suggestion below I renamed the `StringPacksLocaleManager` to `StringPacksLocaleMetaDataProvider`

For example, this file can look like
```
    public class LocaleMetaDataProviderImpl implements StringPacksLocaleMetaDataProvider {
        public String getPackFileIdForLocale(Locale locale) {
            // Applications can return the appropriate pack file id for the locale, based on their mapping in `pack_id_mapping` setting
        }
        
        public boolean shouldAddLanguageAsParentForLocale(Locale locale) {
            // This is a special case because we don't want to add language as a fallback locale for some locales, for example, zh can not be a fallback locale for zh-TW even if they are packed together, because zh is written in a different script than zh-TW.
            // Applications have to handle their special cases and return true or false based on their specific cases
        }
        
        public String getParentLocaleForLocale(Locale locale) {
            // Applications can decide whether to fallback to any specific locale and not necessarily just the language
        }
    }
```